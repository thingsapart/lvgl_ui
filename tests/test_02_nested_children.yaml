# Tests nested indentation with a 'children' block.
- type: obj
  size: [200, 100]
  center: []
  children:
    - type: label
      text: Child Label
      center: []
