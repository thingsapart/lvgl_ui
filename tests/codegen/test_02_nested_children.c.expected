/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // unnamed: obj_0 (obj)
    lv_obj_t* obj_0 = lv_obj_create(parent);

    lv_obj_set_style_size(obj_0, 200, 100, 0);
    lv_obj_center(obj_0);
    // unnamed: label_1 (label)
    lv_obj_t* label_1 = lv_label_create(obj_0);

    lv_label_set_text(label_1, "Child Label");
    lv_obj_center(label_1);


}
